documentation/internal/Level-1-Review-And-LOQ.md

```diff
@@ -0,0 +1,120 @@
+
+## Re-Staring Application
+
+In response to LOQ, an applicant can restart an application. Similar logic to the one described in above section applies.
+
+## Review Decision
+
+Is created for each iteration of a review, can be of status: `'List of questions`, `Conform`, `Non-conform`, `No Decision`. See `Validation of Submission` below for choices that will be available to the user.
+
+## Review Responses Visibility
+
+`is_visible_to_applicant` field (not editable by any user, only by back end) controls visibility of review_response from applicant perspective. And `recommended_applicant_visibility` field (which is editable by reviewer) suggests visibility of review_response to applicant. `updateReviewVisibility` back end action will set `is_visible_to_applicant` when review is submitted. See `LOQ Workflow Diagram` below
+
+## Actions
+
+Front end determines user actions based on review and application status. And theses statuses are changed by back end via actions, here is an abstracted diagram:
+
+![Review-And-LOQ-Actions](images/Review-And-LOQ-Actions.png)
+
+Actions for review are computed by section via `useGetReviewStructureForSections` -> `generateReviewStructure` -> `generateReviewSectionActions`
+
+Actions for applications `changeRequestsProgress` and `isChangeRequest` is used to display actions in Application Home pages, those are generated in `useGetApplicationStructure` -> `addApplicantChangeRequestStatusToElement` and `generateApplicantChangesRequestedProgress`.
+
+See `Structure` below for more details
+
+## Validation of Submission
+
+When **application** is restarted as a result of LOQ, it can only be submitted if general validation passes AND all questions with change requests have been changed. Logic for this is:
+
+latestApplicationResponse is different to previousApplicationResponse where previousApplicationResponse is linked to non confirm reviewResponse
+
+**review** can be submitted as `LOQ` or `Non-conform` when at least review_response is marked as `decline`. Otherwise submission of `Conform` can be made if ALL assigned visible latest application responses have `approve` status review responses.
+
+One thing to note, since we always look at latest application response and it's linked review response, we guarantee that all new changes by applicant will be re-reviewed
+
+## Structure
+
+Application and Review Structure are generated in hooks `useGetApplicationStructure` and `useGetReviewStructureForSection` respectively, these structures are re-computed whenever responses are changed. In the context of Review and LOQ, these fields are added:
+
+### Application
+
+![Applicant LOQ Structure Changes](images/Applicant-LOQ-Structure-Changes.png)
+
+### Review
+
+![Level-1-Review-Structure-Changes](images/Level-1-Review-Structure-Changes.png)
+
+## Components and Hooks
+
+This is described in Application and Review Flow documentation
+
+# LOQ Workflow Diagram
+
+![LOQ Workflow](images/LOQ-Workflow.png)
```

Comment:
Is this related to `recommend_applicant-visibility`?
![Screen Shot 2021-04-22 at 12 22 43 PM](https://user-images.githubusercontent.com/16461988/115637956-839c1200-a365-11eb-9937-14f70c50360a.png)

Also, the bold is half-way through in the first `Visibility field` part of this diagram:
![Screen Shot 2021-04-22 at 12 23 56 PM](https://user-images.githubusercontent.com/16461988/115638015-a1697700-a365-11eb-89a9-4083d206dc8b.png)

Reply:
> Is this related to recommend_applicant-visibility?
yep: 6c760856c386cb9425471d64fbdca35dd048eb99
> Also, the bold is half-way through in the first Visibility field part of this diagram:
6c760856c386cb9425471d64fbdca35dd048eb99